### PR TITLE
WIP, migrate Actions from Skuber to Fabric8

### PR DESCRIPTION
**Pull request by andreaTP**
Sharing for early comments and feedback (don't even know if it works ...)

Quick recap:
 - having separate Fabric8Actions breaks way too many signatures
 - I'm currently relying on some reflection magic, and don't know if I can get rid of it ...
 - having both the skuber and fabric8 clients around is not nice, but is a temporary solution

### CODE REVIEW

File: core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala [hunk 1]
```diff
@@ -208,195 +199,243 @@ abstract class SingleResourceAction[T <: ObjectResource] extends ResourceAction[
...
-        .getOrElse(recoverFromError(client.create(resource), client, nextRetries, executeCreateOrPatch))
+        .map(_ =>
+          recoverFromError(
+            Future {
+              client.secrets().inNamespace().withName()
+              client
+                .resource(resource)
+                .inNamespace(resource.getMetadata.getNamespace)
+                .get()
+                .asInstanceOf[Resource[T, Doneable]] // TODO: this is not going to work ...
```
Comment (andreaTP) [inline core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala:301]: @RayRoestenburg fyi:
This is still a problem, probably `Resource[T, Doneable]` should pop up somewhere in the signature.

File: core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala [hunk 2]
```diff
@@ -208,195 +199,243 @@ abstract class SingleResourceAction[T <: ObjectResource] extends ResourceAction[
...
   override def executing =
     s"Deleting $resourceName resource"
   override def executed =
     s"Deleted $resourceName resource"
 
+  private val log = LoggerFactory.getLogger(this.getClass.getName)
+
+  private val meta = {
+    val default = {
+      Serialization
```
Comment (andreaTP) [inline core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala:405]: @RayRoestenburg fyi:
This looks like a somehow "better way" than using default constructors (this way all the properties injected via Json got populated)

File: core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala [hunk 3]
```diff
@@ -208,195 +199,243 @@ abstract class SingleResourceAction[T <: ObjectResource] extends ResourceAction[
...
+        .readValue("{}", ct.runtimeClass)
+        .asInstanceOf[T]
+    }
+
+    val metadata = new ObjectMetaBuilder()
+      .withName(resourceName)
+      .withNamespace(_namespace)
+      .build()
+
+    default.setMetadata(metadata)
```
Comment (andreaTP) [inline core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala:416]: @RayRoestenburg fyi:
At this point Metadata is empty and we can fill it directly.

**Comment by RayRoestenburg** [comment]
Decided not to do this.

**Pull request closed without merging**